lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -691,11 +707,13 @@ export class BrowserCacheManager extends CacheManager {
     resetRequestCache(state: string): void {
         this.logger.verbose("BrowserCacheManager.resetRequestCache called");
         // check state and remove associated cache items
-        this.getKeys().forEach(key => {
-            if (!StringUtils.isEmpty(state) && key.indexOf(state) !== -1) {
-                this.removeItem(key);
-            }
-        });
+        if (!StringUtils.isEmpty(state)) {
+            this.getKeys().forEach(key => {
+                if (key.indexOf(state) !== -1) {
+                    this.removeItem(key);
+                }
+            });
```

Comment:
Do we need to clear cookies here too?

Reply:
`removeItem` already takes care of it.

Reply:
Right but isn't this the same problem you're fixing down below? It won't reach `removeItem` if `session/localStorage` is empty

Reply:
The issue was that `removeItem` wasn't being reached in `clearRequestByInteractionType`, which isn't the case here. Basically we should never be in a state that `session/localStorage` and cookies are out of sync, which is fixed by this PR.

Reply:
Correct me if I'm wrong: if we call `resetRequestCache` or `clearRequestByState` it will not clear the cookies (aside from the specific 7 listed below) unless there is a corresponding `session/localstorage` entry. This PR currently only addresses `clearRequestByInteractionType`

Reply:
resolved offline, added cookie clearing to cleanRequestByState also.